react-renderer/src/mathtransform/toCustomAD.ts

```diff
@@ -0,0 +1,107 @@
+        "TSUnknownKeyword",
+        "TSVoidKeyword",
+        "TSThisType",
+      ] // from https://github.com/benjamn/ast-types/blob/90a8e63d77fc6134bfd5bfcca793146259219246/def/typescript.ts
+    // transforms binary op to call expression - essentially deconstructs and reconstructs the node
+    const BO2CE = (target: string, node: any) => {
+        const newCallee = j.identifier(target); // get new name of function
+        const newArgs = [node.left, node.right];    // reorder arguments
+        node.callee = newCallee;
+        node.arguments = newArgs;
+        node.type = "CallExpression";
+        return node;
+    };
+    // unary op to call expression
+    const UO2CE = (target: string, node: any) => {
+        const newCallee = j.identifier(target); // get new name of function
+        const newArgs = [node.argument];    // reorder arguments
+        node.callee = newCallee;
+        node.arguments = newArgs;
+        node.type = "CallExpression";
+        return node;
+    };
+    // type preset keyword (number, etc.) to type reference i.e. custom type OR custom type to custom type
+    // TS parser treats these differently, so it would have been prettier to separate the two, but for ease of future use I allowed them to be treated the same
+    const TYPSUB = (target: string, node: any) => {
+        const newTName = j.identifier(target);
+        node.type = "TSTypeReference";
+        node.typeName = newTName;
+        return node;
+    }
+
+    const BOPS : {[key:string]:any[]} = {
+        "+": ["add", BO2CE],    // [new name, structure to convert to]
+        "-": ["sub", BO2CE],    // supports custom structure transformations - e.g. not all bops get transformed to call expressions
+        "*": ["mul", BO2CE],    // if you don't want to transform a specific operation at all just write a basic function instead of bo2ce that returns the node
+        "/": ["div", BO2CE]
+    }; 
+    const UOPS : {[key:string]:any[]} = {
+        "-": ["neg", UO2CE]
+    }
+    const TYPS : {[key:string]:any[]} = {
+        // "Tensor": ["VarAD", TYPSUB],   // TYPSUB - method to perform type substitution. This will practically always be the method used to transform any type. 
+        "TSNumberKeyword": ["VarAD", TYPSUB],
+        // "OLDTYPE": ["NEWTYPE", TYPSUB]
+        
+    }
+    // find how a node should be translated and return the translated node
+    const transNodes = (nodes: any) => {
+        nodes.replaceWith((nodePath: any) => {
+            const {node} = nodePath;
+            const transformData = getTransformData(node);
+
+            // the below line is theoretically bad practice per https://stackoverflow.com/questions/1098040/checking-if-a-key-exists-in-a-javascript-object
+            // however, given that our opmaps are manually generated it is low-risk. 
+            return transformData ? transformData[1](transformData[0], node) : node;
+        })
+    };
+    const MARKTAG = "mathtrans";
```

Comment:
It will actually transform the nearest AST node below the marker. If that's a function, or a statement, or something else.  It checks the LeadingComments node that the parser attaches to another AST node, which I believe is whatever the main type of a line is (method call, function, variable declaration, etc.).

Sample Input:
```
export const objDict = {
    above: (x: number[]): number => {
      //autodiff
      x = t2 - z * 34 + (2 / 4)
      y = a + bigint
    }
}
```
Sample Output:
```
export const objDict = {
    above: (x: number[]): number => {
      //autodiff
      x = add(sub(t2, mul(z, 34)), div(2, 4))
      y = a + bigint
    }
}
```

Renamed in 635db53